src/App.jsx

```diff
@@ -1,58 +1,117 @@
 import 'bulma/css/bulma.css';
 import './App.scss';
+import React, { useState } from 'react';
+import cn from 'classnames';
 
 export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
+  { id: 1, name: 'Dumplings' },
+  { id: 2, name: 'Carrot' },
+  { id: 3, name: 'Eggs' },
+  { id: 4, name: 'Ice cream' },
+  { id: 5, name: 'Apple' },
+  { id: 6, name: 'Bread' },
+  { id: 7, name: 'Fish' },
+  { id: 8, name: 'Honey' },
+  { id: 9, name: 'Jam' },
+  { id: 10, name: 'Garlic' },
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const ARRAY_METHOD = 'reverse';
+
+function goodsSorter(goods, { sortType, arrayMethod }) {
+  const sortedGoods = [...goods];
+
+  if (sortType) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SORT_BY_ALPHABET:
+          return good1.name.localeCompare(good2.name);
+        case SORT_BY_LENGTH:
+          return good1.name.length - good2.name.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (arrayMethod) {
+    sortedGoods[arrayMethod]();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [arrayMethod, setArrayMethod] = useState('');
```

Comment:
let's save boolean value `isReversed` instead of method string

Reply:
Not resolved.